### ISSUE TITLE
Fault not parsed correctly

### ISSUE THREAD

**Issue by gettalong** [unwire/handsoap#6]
Hi,

in the method Fault.from_xml in lib/handsoap/service.rb the namespace selector is partially missing, ie. instead of 

```
fault_code = node.xpath('./faultcode', ns).to_s
```

it should probably be

```
fault_code = node.xpath('./env:faultcode', ns).to_s
```

Same with `faultstring` and `detail`.

Best regards,
  Thomas

**Comment by jimmiw** [comment]
Hello,

After looking at http://schemas.xmlsoap.org/soap/envelope/ it seems that the current implementation is ok.

Remember that all this is depending on you using SOAP1.1 not SOAP1.2 (SOAP1.2 specs: http://www.w3.org/TR/soap12-part1/#soapfault)

**Comment by gettalong** [comment]
Hmm... I'm not good at reading XSD files but where does it say that the child elements `faultcode`, `faultstring` and `detail` of `Fault` don't live in the same namespace? For example, is the following not a vaild fault response:

```
<?xml version='1.0' encoding='UTF-8'?>
<a:Envelope xmlns:a='http://schemas.xmlsoap.org/soap/envelope/'>
  <a:Body>
    <a:Fault>
      <a:faultcode>a:Client</a:faultcode>
      <a:faultstring>Failed to parse the request</a:faultstring>
      <a:detail>Invalid parameter</a:detail>
    </a:Fault>
  </a:Body>
</a:Envelope>
```

This is what is returned by an Intel AMT 5.0 device. As you can see, the child elements are properly namespaced. But since the current implementation only checks for unnamespaced child elements, the correct information is not read out. For example, you have

```
fault_code = node.xpath('./faultcode', ns).to_s
```

and explicitly specify the namespace as second argument to the `#xpath` method but you don't use the defined namespace `env` in the first argument. This seems odd...

Could you shed some light on this?

**Comment by jimmiw** [comment]
The SOAP returned, from your INTEL machine, seems fine.
I'm currently making some tests for this issue and getting some more qualified people to look at my solution. 

Just to let you know that there is progress on the issue.

**Comment by gettalong** [comment]
Thanks! Just so you know: when changing the path to include the env namespace, ie. `./faultcode` to `./env:faultcode`, the faults from my test machine are reported correctly.

**Comment by jimmiw** [comment]
True, but the tests fail.

**Comment by jimmiw** [comment]
This is now fixed in version 1.1.2

**Comment by gettalong** [comment]
Thanks!
